src/Language/Mimsa/Types/Project/Project.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Language.Mimsa.Types.Project.Project where
+
+import qualified Data.Aeson as JSON
+import Data.Text (Text)
+import GHC.Generics (Generic)
+import Language.Mimsa.Types.Identifiers (Name, TyCon)
+import Language.Mimsa.Types.Project.VersionedMap (VersionedMap)
+import Language.Mimsa.Types.Store (ExprHash, Store)
+
+------
+
+newtype ServerUrl = ServerUrl {getServerUrl :: Text}
+  deriving (Eq, Ord, Show)
+  deriving newtype (JSON.ToJSON, JSON.FromJSON)
+
+-------
+
+-- our environment contains whichever hash/expr pairs we have flapping about
```

Comment:
Although I hate busy work for the sake of it, should probably break this file up into separate types for consistency.